components/automate-cli/cmd/chef-automate/cleanup.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net"
+	"strings"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+)
+
+var cleanupFlags = struct {
+	onprem        bool
+}{}
+
+func init() {
+	RootCmd.AddCommand(cleanupCmd)
+	cleanupCmd.PersistentFlags().BoolVar(&cleanupFlags.onprem, "onprem", false, "Cleaning up all the instances related to onprem ")
+
+}
+
+var cleanupCmd = &cobra.Command {
+		Use:  "cleanup",
+	    Short: "cleanup the Automate HA instances",
+	    Long:  "cleaning up the instance of all the Automate HA related Applications.",
+		Annotations: map[string]string{
+			NoCheckVersionAnnotation: NoCheckVersionAnnotation,
+		},
+		RunE: runCleanupCmd,
+	}
+
+const (
+		FRONTENDCLEANUP_COMMANDS = `
+		sudo systemctl stop chef-automate;
+		sudo rm -rf /hab;
+		`
+	
+		BACKENDCLEANUP_COMMANDS = `
+		sudo systemctl stop hab-sup;
+		sudo rm -rf /hab;   
+		`
+)
+
+
+
+func runCleanupCmd(cmd *cobra.Command, args []string) error {
+	infra, err := getAutomateHAInfraDetails()
+	if err != nil {
+		return err
+	}
+	sshUser := infra.Outputs.SSHUser.Value
+	sskKeyFile := infra.Outputs.SSHKeyFile.Value
+	sshPort := infra.Outputs.SSHPort.Value
+	writer.Printf(strings.Join(args, ""))
+	if isA2HARBFileExist() {
+		if cleanupFlags.onprem {
+			frontendIps := append(infra.Outputs.ChefServerPrivateIps.Value, infra.Outputs.AutomatePrivateIps.Value...)
+			backendIps := append(infra.Outputs.PostgresqlPrivateIps.Value, infra.Outputs.OpensearchPrivateIps.Value...)
+			for i := 0; i < len(frontendIps); i++ {
+				executeCleanupOnRemote(sshUser, sshPort, sskKeyFile, frontendIps[i], FRONTENDCLEANUP_COMMANDS)
+			}
+			for i := 0; i < len(backendIps); i++ {
+				executeCleanupOnRemote(sshUser, sshPort, sskKeyFile, backendIps[i],BACKENDCLEANUP_COMMANDS)
+			}
+			cleanUpScript := "hab pkg uninstall chef/automate-ha-deployment"
+			args := []string{
+				"-c",
+				cleanUpScript,
+			}
+			err := executeCommand("/bin/sh", args, "")
+			if err != nil {
+				return err
+			}
+			return nil	
+		}
+		
+	}
+	return nil
+}
+
+func executeCleanupOnRemote(sshUser string, sshPort string, sshKeyFile string, ip string, commands string) {
```

Comment:
make this function more generic to execute commands in remote machine, and check if we have similar code for the same then use it